vpr/src/route/route_common.cpp

```diff
@@ -713,24 +690,49 @@ float get_rr_cong_cost(int inode) {
                     continue; //Already included above
                 }
 
-                cost += get_single_rr_cong_cost(node);
+                cost += get_single_rr_cong_cost(node, pres_fac);
             }
         }
     }
     return (cost);
 }
 
-/* Returns the congestion cost of using this rr_node, *ignoring*
- * non-configurable edges */
-static float get_single_rr_cong_cost(int inode) {
+/* Returns the congestion cost of using this rr_node,
+ * *ignoring* non-configurable edges */
+float get_single_rr_cong_cost(int inode, float pres_fac) {
+    return get_single_rr_cong_base_cost(inode) *
+            get_single_rr_cong_acc_cost(inode) *
+            get_single_rr_cong_pres_cost(inode, pres_fac);
+}
+
+/* Returns the base cost of using this rr_node */
+float get_single_rr_cong_base_cost(int inode) {
     auto& device_ctx = g_vpr_ctx.device();
+    auto cost_index = device_ctx.rr_nodes[inode].cost_index();
+
+    return device_ctx.rr_indexed_data[cost_index].base_cost;
+}
+
+/* Returns the accumulated congestion cost of using this rr_node */
+float get_single_rr_cong_acc_cost(int inode) {
```

Comment:
Explore if putting inline on these helps, or pasting the function body into the calling function.  get_single_rr_cong_cost is going to be a very hot function, and it calls these other short functions.  Good coding style as long as they are inlined, but worth checking speed (or you could look at the assembly or run the optimized code through the debugger to see if they are inlined).